### ISSUE TITLE
Wii U WUHB

### ISSUE THREAD

**Issue by Ducky-Duck-Duck** [Wohlstand/TheXTech#665]
Please compile the Wii U version as a Wii U Homebrew Bundle(WUHB) as RPX files are outdated and many no long work with the new standard enviroment Aroma (this can be done using [WUT](https://github.com/devkitPro/wut))

**Comment by ds-sloth** [comment]
It's nice that wuhb can be made with the normal devkitPro tools.

Are there any users (emulator users, users of older homebrew environments, etc) who can use rpx and cannot use wuhb?

**Comment by Ducky-Duck-Duck** [comment]
> It's nice that wuhb can be made with the normal devkitPro tools.
> 
> Are there any users (emulator users, users of older homebrew environments, etc) who can use rpx and cannot use wuhb?

Yes, users of Tiramisu or Haxchi (older environments) cannot run wuhb, although I am not sure about emulation. The Wii U community is trying to kill off Tiramisu because Aroma is (im pretty sure) essentially better in every way, and Haxchi/CBHC is completely unsupported.

**Comment by Ducky-Duck-Duck** [comment]
oops^

**Comment by Ducky-Duck-Duck** [comment]
> Yes, users of Tiramisu or Haxchi (older environments) cannot run wuhb, although I am not sure about emulation. 

Many apps (Most?) are released with a wuhb and an rpx for convenience

**Comment by ds-sloth** [comment]
I just added a WUHB build to our CI (in addition to the existing RPX build). Can you please check whether the resulting package works fine on your console?

**Comment by Ducky-Duck-Duck** [comment]
I'll do so when I get home(3-4 hrs) tysm

**Comment by Wohlstand** [comment]
Building of WUHB I already explained in the manual as a way to improve the boot performance (by embedding all resources into one WUHB image), etc. CEMU doesn't support WUHB playing yet.
I'm not sure, will resourceless WUHB work normally or will fail... the hardware is needed.

**Comment by Ducky-Duck-Duck** [comment]
> I just added a WUHB build to our CI (in addition to the existing RPX build). Can you please check whether the resulting package works fine on your console?

I am unable to find it, I'm most likely just dumb but could you send me a link perhaps?

**Comment by Ducky-Duck-Duck** [comment]
Alright, found it, downloaded it, and put it on my SD card. There is an issue with the name on the home menu, it's ‘TheXTech v$THEXTECH_VERSION_STRING’. Also when you boot it, it launches, sits on the booting screen for a bit then goes black. I'll edit this comment if it changes

**Comment by ds-sloth** [comment]
Can you please test the latest version once the CI has finished?

**Comment by Wohlstand** [comment]
Also, a small note: WUHB may not boot because it finds presense of `fs:/vol/content/` virtual path but no packed resources inside. I think, there is also needed to make the more stricter validator of the detected directory rather than just checking for its existance here: https://github.com/Wohlstand/TheXTech/blob/main/lib/AppPath/private/app_path_wiiu.cpp#L57-L119 (Adding a kind of a static call `s_verifyDirectory(const std::string&)` that will check presense of typical directories and files like graphics, music.ini, sounds.ini, and gameinfo.ini).

**Comment by ds-sloth** [comment]
I made a slightly more strict validator to check for the presence of `fs:/vol/content/graphics/`.

> Initially, supposed that WUHB should be created with packing of resources, and do WUHB packing outside CMake build, with adding of SMBX and AoD assets separately, re-using the same RPX binary.

I see, that is much more efficient. We could use the CMake-built WUHB as a "plain assets" package, and add steps to the CI workflow to package extra WUHBs with the each of the main asset packs?

**Comment by Wohlstand** [comment]
> add steps to the CI workflow to package extra WUHBs with the each of the main asset packs?

Ye!

**Comment by ds-sloth** [comment]
I can add that to the CI workflow. Do you want to use separate icons / splash screens for those? (Do they already exist?)

**Comment by Wohlstand** [comment]
> Do you want to use separate icons / splash screens for those? (Do the asset-specific icons / splashes already exist?)

Would be better, then just add them to assets packs, and I will sync them up to the mainstream.
